content/docs/intro/concepts/project.md

```diff
@@ -144,12 +118,56 @@ var myTask = new Task("myTask", new TaskArgs
 
 {{< /chooser >}}
 
+## Getting the Current Project Programmatically
+
+The {{< pulumi-getproject >}} function gives you the currently deploying project. This can be useful for naming or tagging resources.
+
+{{< chooser language "javascript,typescript,python,go,csharp" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+let project = pulumi.getProject();
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+let project = pulumi.getProject();
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+project = pulumi.get_project()
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+project := ctx.Project()
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var project = Deployment.Instance.ProjectName;
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
 ## Stack Settings Files {#stack-settings-file}
 
-Each stack that is created in a project will have a file named `Pulumi.<stackname>.yaml` which contains the configuration specific to this stack.
+Each stack that is created in a project will have a file named `Pulumi.<stackname>.yaml` that contains the configuration specific to this stack. This file typically resides in the root of the project directory.
 
 For stacks that are actively developed by multiple members of a team, the recommended practice is to check them into source control as a means of collaboration. Since secret values are encrypted, it is safe to check in these stack settings.
 
-For stacks that are ephemeral or are used in "inner loop" development, the stack settings are typically not checked into source control.
+When using ephemeral stacks, the stack settings are typically not checked into source control.
```

Comment:
What are "ephemeral stacks"? Is that a feature of Pulumi? I'm not sure this sentence (or the previous version) is adding much value; the distinction between ephemeral/persistent doesn't seem relevant or important here, unless we add some information as to _why_ it may not be useful to check in stack-level config for ephemeral stacks.

Reply:
Yup, I'll revisit this in a follow-up issue.